src/directiveSelectorRule.ts

```diff
@@ -0,0 +1,143 @@
+        this.setMultiPrefix(prefix);
+        this.setPrefixArguments(prefix);
+        this.setPrefixValidator(prefix);
+        this.setPrefixFailure();
+        if (name === 'camelCase') {
+            this.nameValidator = SelectorValidator.camelCase;
+        }
+    }
+
+    private setMultiPrefix(prefix:string){
+        this.isMultiPrefix = typeof prefix==='string';
+    }
+
+    private setPrefixArguments(prefix:any){
+        this.prefixArguments = this.isMultiPrefix?prefix:prefix.join(',');
+    }
+
+    private setPrefixValidator(prefix:any){
+        let prefixExpression: string = this.isMultiPrefix?prefix:(prefix||[]).join('|');
+        this.prefixValidator = SelectorValidator.multiPrefix(prefixExpression);
+    }
+
+    private setPrefixFailure(){
+        this.FAILURE_PREFIX = this.isMultiPrefix?FAILURE_PREFIX_SINGLE:FAILURE_PREFIX_MANY;
+    }
+
+    public validateName(attrs:string[]):boolean {
+        return attrs.some((a)=>this.nameValidator(a));
+    }
+
+    public validatePrefix(attrs:string[]):boolean {
+        return attrs.some((a)=>this.prefixValidator(a));
+    }
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        let documentRegistry = ts.createDocumentRegistry();
+        let languageServiceHost = Lint.createLanguageServiceHost('file.ts', sourceFile.getFullText());
+        let languageService : ts.LanguageService = ts.createLanguageService(languageServiceHost, documentRegistry);
+        return this.applyWithWalker(
+            new SelectorValidatorWalker(
+                sourceFile,
+                languageService,
+                this));
+    }
+
+    public getFailureString(failureConfig): string {
+        return sprintf(failureConfig.fail, failureConfig.className, this.getOptions().ruleArguments, failureConfig.selector);
+    }
+
+}
+
+class SelectorValidatorWalker extends Lint.RuleWalker {
+    private languageService : ts.LanguageService;
+    private typeChecker : ts.TypeChecker;
+
+    constructor(sourceFile: ts.SourceFile, languageService : ts.LanguageService, private rule: Rule) {
+        super(sourceFile, rule.getOptions());
+        this.languageService = languageService;
+        this.typeChecker = languageService.getProgram().getTypeChecker();
+    }
+
+    visitClassDeclaration(node: ts.ClassDeclaration) {
+        (<ts.Decorator[]>node.decorators || [])
+            .forEach(this.validateDecorator.bind(this, node.name.text));
+        super.visitClassDeclaration(node);
+    }
+
+    private validateDecorator(className: string, decorator: ts.Decorator) {
+        let baseExpr = <any>decorator.expression || {};
+        let expr = baseExpr.expression || {};
+        let name = expr.text;
+        let args = baseExpr.arguments || [];
+        let arg = args[0];
+        if (name === 'Directive') {
+            this.validateSelector(className, arg);
+        }
+    }
+
+    private validateSelector(className: string, arg: ts.Node) {
```

Comment:
Same comment as above. This can be simplified by filtering only properties with supported types.